test/timeline.js

```diff
@@ -0,0 +1,203 @@
+      } else {
+        eventTypeId = 123
+        return nockHelper.mockPostOauthEndpoint(
+          `/integrations/v1/${applicationId}/timeline/event-types/123/properties`,
+          {
+            name: 'NumericProperty',
+            label: 'Numeric Property',
+            propertyType: 'Numeric',
+          },
+          { userId },
+        )()
+      }
+    })
+    afterEach(nockHelper.resetNock)
+
+    it('should create an event type property', async () => {
+      return hubspot.timelines
+        .createEventTypeProperty(eventTypeId, applicationId, userId, {
+          name: 'NumericProperty',
+          label: 'Numeric Property',
+          propertyType: 'Numeric',
+        })
+        .then(data => {
+          expect(data.name).to.eq('NumericProperty')
+        })
+    })
+  })
+
+  describe('updateEventTypeProperty', () => {
+    let eventTypeId
+    let eventTypePropertyId
+    beforeEach(() => {
+      if (process.env.NOCK_OFF) {
+        return createEventType().then(data => {
+          eventTypeId = data.id
+          return createEventTypeProperty(eventTypeId).then(
+            data => (eventTypePropertyId = data.id),
+          )
+        })
+      } else {
+        eventTypeId = 123
+        eventTypePropertyId = 234
+        return nockHelper.mockPutOauthEndpoint(
+          `/integrations/v1/${applicationId}/timeline/event-types/123/properties`,
+          {
+            name: 'NumericProperty',
+            label: 'A new label',
+            propertyType: 'Numeric',
+            id: eventTypePropertyId,
+          },
+        )()
+      }
+    })
+    afterEach(nockHelper.resetNock)
+
+    it('should update an event type property', async () => {
+      return hubspot.timelines
+        .updateEventTypeProperty(
+          eventTypeId,
+          applicationId,
+          eventTypePropertyId,
+          {
+            name: 'NumericProperty',
+            label: 'A new label',
+            propertyType: 'Numeric',
+          },
+        )
+        .then(data => {
+          expect(data.label).to.eq('A new label')
+        })
+    })
+  })
+
+  describe('createTimelineEvent', () => {
+    let eventTypeId
+    beforeEach(() => {
+      if (process.env.NOCK_OFF) {
+        return createEventType().then(data => (eventTypeId = data.id))
+      } else {
+        eventTypeId = 123
+        return nockHelper.mockFuzzyPutOauthEndpoint(
+          `/integrations/v1/${applicationId}/timeline/event`,
+          body =>
+            !!body.id &&
+            body.email === 'test@test.com' &&
+            body.eventTypeId === eventTypeId,
+          undefined,
+        )()
+      }
+    })
+    afterEach(nockHelper.resetNock)
+
+    it('should create an event', async () => {
+      return hubspot.timelines
+        .createTimelineEvent(applicationId, eventTypeId, {
+          email: 'test@test.com',
+        })
+        .then(data => {
+          expect(data).to.be.an('undefined')
```

Comment:
I'm a bit uncertain about why the data is `undefined`. I imagine we would check the response to verify a TimelineEvent with the email `test@test.com` has been created.

I see in the docs that the email is used as an identifier. In this case, the email matches the contact "you're adding the event to". When you ran this against the API for realz, did the API error since there's no contact with the email "test@test.com"?

Reply:
Such a good question! From the [documentation](https://developers.hubspot.com/docs/methods/timeline/create-or-update-event):

> Returns a 204 response on success. Otherwise, you'll receive a 4xx error, with more details about the specific error in the body of the response.

So on success the body is empty or `undefined`